src/main/java/com/logtail/logback/LogtailAppender.java

```diff
@@ -83,23 +85,40 @@ protected void append(ILoggingEvent event) {
             return;
 
         if (this.ingestUrl.isEmpty() || this.sourceToken == null || this.sourceToken.isEmpty()) {
-            errorLog.warn("Missing Source token for Better Stack - disabling LogtailAppender. Find out how to fix this at: https://betterstack.com/docs/logs/java ");
+            // Prevent potential dead-lock, when a blocking logger is configured - avoid using errorLog directly in append
+            startThread("logtail-warning-logger", () -> {
+                errorLog.warn("Missing Source token for Better Stack - disabling LogtailAppender. Find out how to fix this at: https://betterstack.com/docs/logs/java ");
+            });
             this.disabled = true;
             return;
         }
 
-        batch.add(event);
+        if (batch.size() < maxQueueSize) {
+            batch.add(event);
+        }
+
+        if (warnAboutMaxQueueSize && batch.size() == maxQueueSize) {
+            this.warnAboutMaxQueueSize = false;
+            // Prevent potential dead-lock, when a blocking logger is configured - avoid using errorLog directly in append
+            startThread("logtail-error-logger", () -> {
+                errorLog.error("Maximum number of messages in queue reached ({}). New messages will be dropped.", maxQueueSize);
+            });
```

Comment:
This was a bit unintuitive, but there was an issue with the `AsyncAppender` (from our default config) blocking at this point (since it's queue was full) which ended up dead-locking the main thread.

It can be avoided by reconfiguring the `AsyncAppender` (by increasing its `queueSize` or setting `neverBlock` to `true`), but the configuration would have to be changed by upgrading users - and I really don't want to risk a dead-lock.

I'll create a separate task for removing `AsyncAppender` from the recommended config altogether - I think there's no need for it anymore (it was probably introduced before we flushed the queue in a separate thread).